src/main/java/aquality/selenium/core/waitings/ConditionalWait.java

```diff
@@ -0,0 +1,104 @@
+package aquality.selenium.core.waitings;
+
+import aquality.selenium.core.applications.IApplication;
+import aquality.selenium.core.configurations.ITimeoutConfiguration;
+import aquality.selenium.core.localization.ILocalizationManager;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BooleanSupplier;
+
+public class ConditionalWait implements IConditionalWait {
+
+    private IApplication application;
+    private ITimeoutConfiguration timeoutConfiguration;
+    private ILocalizationManager localizationManager;
+
+    @Inject
+    public ConditionalWait(Provider<IApplication> applicationProvider, ITimeoutConfiguration timeoutConfiguration, ILocalizationManager localizationManager) {
+        application = applicationProvider.get();
+        this.timeoutConfiguration = timeoutConfiguration;
+        this.localizationManager = localizationManager;
+    }
+
+    @Override
+    public boolean waitFor(BooleanSupplier condition, String message) {
+        return waitFor(condition, timeoutConfiguration.getCondition(), timeoutConfiguration.getPollingInterval(), message);
+    }
+
+    @Override
+    public boolean waitFor(BooleanSupplier condition, long timeoutInSeconds, long pollingIntervalInMilliseconds, String message) {
+        try {
+            waitForTrue(condition, timeoutInSeconds, pollingIntervalInMilliseconds, message);
+            return true;
+        } catch (TimeoutException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public void waitForTrue(BooleanSupplier condition, String message) throws TimeoutException {
+        waitForTrue(condition, timeoutConfiguration.getCondition(), timeoutConfiguration.getPollingInterval(), message);
+    }
+
+    @Override
+    public void waitForTrue(BooleanSupplier condition, long timeoutInSeconds, long pollingIntervalInMilliseconds, String message) throws TimeoutException {
```

Comment:
we should have exceptions to ignore as optional parameter in every method of the ConditionalWait. 
I remember we done it in dotnet (https://github.com/aquality-automation/aquality-selenium-core-dotnet/issues/49) and you agreed that it would be better

Reply:
outdated, sorry